### ISSUE TITLE
DOCS suggestion

### ISSUE THREAD

**Issue by michaelaye** [intake/intake#368]
Could you post a simple step-by-step example for a remotely stored file on a Dropbox location?
The current documentation is still too hard for me to be able to create a dataset.

[This](https://intake.readthedocs.io/en/latest/data-packages.html) mentions I should run `conda-build` to create the skeleton files, but there doesn't seem to be a way to run `Conda-skeleton` without pointing to existing packages?

Also, this sentence in https://intake.readthedocs.io/en/latest/data-packages.html:

> The ``$SRC_DIR`` variable refers to any source tree checked out (from Github or other service), and the
``$RECIPE_DIR`` refers to the directory where the ``meta.yaml`` is located.

makes it sound like a simple remote URL could not be used for creating a package?

**Comment by michaelaye** [comment]
This post came to the rescue:

https://www.informaticslab.co.uk/home/2018/9/13/tutorial-how-to-build-an-intake-catalog

Now next question is, how to structure the yaml file to have more than one data file in the dataset.

**Comment by martindurant** [comment]
Given the other issue (#366 ) was fixed, is this still an issue?

If yes, I'd appreciate a little more detail about what you want to achieve. The simplest way to share the details of a given dataset is to write the spec to a file (e.g., from the source's `.yaml()` method) and put that file into a sharable location.

If you actually want a conda package, then the linked page is indeed the right place to look. You would want to do this if one of the following applies:
- you need to specify the dependencies of the data
- you want to make use of conda's versioning and trusted channels; maybe it is the only permitted way for people to get data
- you want to include the data itself and the spec to read it in one go, which you might for small reference data-sets
- you want to include code along with the data specification
I would like to think that the page provides good instructions, but if you are confused, it clearly could be better! Perhaps the first point: yes, the catalog file in a conda package can refer to data supplied with the package, in which case you would need to use the special environment variables, but referring to fixed remote URLs is totally allowed, e.g., https://github.com/intake/intake-examples/tree/master/airline_flights

**Comment by martindurant** [comment]
That I can answer immediately:
```
sources:
  car_park_tickets_sold:
    description: Data about the number of tickets sold in Exeter car parks from Exeter City Council (https://exeterdatamill.com/dataset/car-park-tickets-sold)
    driver: csv
    ...
  next_data_set:
    description: more stuff
```
This is just YAML syntax, you add more blocks with indentation showing the thing they belong in. Sources is a list of named blocks, each of which is a data source (i.e., it works like a dictionary).

**Comment by michaelaye** [comment]
Yes, I'm interested in the conda package because of the versioning and how a user can simply do `conda update <dataset>` to get the latest version.

From above, it seems that both `car_park_tickets_sold` and `next_data_set` would appear in the root level of the catalog of `intake`. Upon digging into the docs, it seems possible either by using $CATALOG_DIR or by nesting of catalog files (not sure I grok the difference) to access several data files as part of a data set?

My goal is to provide several data files as part of one conda package, e.g. named `p4data`.
If possible, I'd like to access the subfiles by their names under the namespace `p4data`, for example, for a dataset called `fans`:

```python
df = intake.cat.p4data.fans.read()
```
I would believe, maybe wrongly and egocentrically, that providing a bunch of files under a unifying name-space AND under the versioning of conda would be the prime use case for intake, so seeing a full example for that would have made things much simpler, but at least I am convinced now that it's somehow possible! ;)

**Comment by martindurant** [comment]
I am tempted to ask you to have a go, so I can have a clearer idea of what is confusing you. Obviously, the documentation should then be updated to make these things clearer. There is quite an amount of documentation already, but Intake comes with many concepts that are novel for many pydata users.

> it seems that both car_park_tickets_sold and next_data_set would appear in the root level of the catalog

Correct

> df = intake.cat.p4data.fans.read()

This would be nesting, entry `p4data` would be a type of `Catalog`, such as `YAMLFileCatalog`.

@jsignell (if you are around), do you have a moment to make a conda-installable catalog on intake-examples, that has the other example catalogs as its items?

**Comment by michaelaye** [comment]
I think everything is in the docs, but it's located at different places, so it's quite time consuming to put everything together in my head. ;) For example, how the YAML file exactly affects how I access the data isn't clear to read on one page, I think. I always learn best by examples, so to have an example for a conda-package with several subfiles would definitely be a nice thing to follow. 
Thanks for letting me know that this is nesting, I wasn't so sure still.

**Comment by michaelaye** [comment]
It doesn't seem to work as suggested.

Here are the contents of my files:

meta.yaml:

```yaml
$ cat meta.yaml                                                                              (py37)
# meta.yaml
package:
  version: '1.1.2'
  name: 'planetfour-data'

build:
  number: 0
  noarch: generic

requirements:
  run:
    - intake
  build: []

about:
  description: The Planet Four catalog (https://www.planetfour.org/results)
  license: OGL v3
  license_family: OTHER
  summary: Planet Four Catalog
```

p4data.yaml:
```yaml
sources:
  p4data:
    args:
      path: ./p4data_sub.yaml
    description: ''
    driver: intake.catalog.local.YAMLFileCatalog
    metadata: {}
```

p4data_sub.yaml:
```yaml
# p4data_fans.yaml
sources:
  fans:
    description: Fan data of the Planet Four catalog
    driver: csv
    args:
      urlpath: 'https://www.dropbox.com/s/o70a1x8xzrxfyvg/P4_catalog_v1.1_L1C_cut_0.5_fan.csv.gz?dl=1'
      csv_kwargs:
        blocksize: null
        compression: gzip
    metadata:
      origin_url: 'https://www.planetfour.org/results'
  blotches:
    description: Blotch data of the Planet Four catalog
    driver: csv
    args:
      urlpath: 'https://www.dropbox.com/s/hx0qr1ut5lk2fva/P4_catalog_v1.1_L1C_cut_0.5_blotch.csv.gz?dl=1'
      csv_kwargs:
        blocksize: null
        compression: gzip
    metadata:
      origin_url: 'https://www.planetfour.org/results'
```

build.sh:
```bash
#!/bin/bash

mkdir -p $PREFIX/share/intake/p4data
cp $RECIPE_DIR/p4data*.yaml $PREFIX/share/intake/
```

but all files are still in the root catalog after installing:

```python
In [6]: list(intake.cat)
Out[6]: ['p4data', 'fans', 'blotches']
```

**Comment by martindurant** [comment]
Something like
```bash
mkdir -p $PREFIX/share/intake/p4data
cp $RECIPE_DIR/p4data.yaml $PREFIX/share/intake/  # only the master cat
cp $RECIPE_DIR/p4data_sub.yaml $PREFIX/share/intake/p4data  # should *not* be in top dir
```
and the source
```
sources:
  p4data:
    args:
      path: "{{ CATALOG_DIR }}/p4data/p4data_sub.yaml"
    description: ''
    driver: intake.catalog.local.YAMLFileCatalog
    metadata: {}
```

explanations:
- intake will load all yaml files in the share/intake/ directory and put all of their entries in the top level
- the canonical way to include extra stuff is in a directory; you had made one but put nothing in it
- the path to the cat should not be relative (that would be from the *current* directory) but use the special template variable meaning "the directory in which this cat is"

**Comment by michaelaye** [comment]
hm, `mkdir -p $PREFIX/share/intake/p4data` seems to fail?

**Comment by michaelaye** [comment]
btw, I copied that wrong build.sh from the above linked blog post! ;)

```
mkdir -p $PREFIX/share/intake/exeter-car-park-tickets-sold
cp $RECIPE_DIR/car-park-tickets-sold.yaml $PREFIX/share/intake/
```

**Comment by martindurant** [comment]
Did you solve things? If `mkdir -p ...` fails, there ought to be an error message. Since this is running within a conda environment, that location should be writable.

**Comment by michaelaye** [comment]
`$PREFIX` needs to be `$CONDA_PREFIX`, then it works.

Here's a repo with everything required to create the planet four catalog:

https://github.com/michaelaye/p4data

**Comment by martindurant** [comment]
Excellent! Would you be interested in describing your experience and/or helping us improve the documentation? In fact, we could include your conda package in intake-examples, if you like.

**Comment by michaelaye** [comment]
I realized that I had the wrong name because we distinguish between the project's data and the result catalog, so I had to rename above repository to p4catalog:

https://github.com/michaelaye/p4catalog

I added a hopefully adequate README to my repo, feel free however you see fit to copy the example or link to it. Thanks again for your help in getting this going, I'm excited to offer this now to my students as the easiest way to get our data files!
I can also edit an .rst file in your docs, if you prefer, just point me to where you want it in that case.
